open-craft/edx-platform
# LX-84 Make CAPA work in studio

The purpose of this PR is to:
- [ ] Get Capa XBlock editing working in Studio
- [ ] Ensure course import/export works for legacy course data

**Dependencies**
* https://github.com/edx/xblock-utils/pull/52

**Testing instructions**:

1. Run this branch in your devstack.
1. Run `make studio-static` and `make lms-static` to rebuild static assets.
1. In Studio, ensure that you can add and edit CAPA problems as usual.
1. Test the Simple (markdown) editor:
   * The buttons across the top bar should add markdown, and the cheatsheet should toggle towards the right side of the editor.
   * Saving changes should persist to the OLX data.
1. Test Advanced editor -- once a problem is changed to Advanced editing, it should remain editable only from the Advanced editor.
1. Import the [edx-demo-course](https://github.com/edx/edx-demo-course/releases/tag/open-release%2Fhawthorn.2) into a new course.
   Ensure that the CAPA problems import without issue.
1. Export the demo course, and re-import using that export.
   Ensure that the CAPA problems still work.

**Author notes and concerns**:

1. Need to handle LaTeX source editor.
1. Do we need to do anything to support complex capa problems, e.g. Chemical Equations and Molecule editors?
1. Failing tests will be addressed in the main feature PR.

**Reviewers**
